From 3b192612d9b22e10ac1d95a481c0ad8425ad2c23 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 22 Jun 2019 12:07:06 +0530
Subject: [PATCH 304/386] octeontx2-af: Create tx_stall debugfs entry only for
 96xx

Silicons otherthan 96xx A0 doesn't have HW issues which result
in transmit stall, hence don't create tx_stall debugfs entry.

Change-Id: I9903d344634c6fac06cafd6315b2ffac93b49d42
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11460
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/rvu_debugfs.c   | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
index 0ed4b3516683..4f34d3b07edd 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
@@ -1429,10 +1429,13 @@ static void rvu_dbg_nix_init(struct rvu *rvu)
 	if (!pfile)
 		goto create_failed;
 
-	pfile = debugfs_create_file("tx_stall_hwissue", 0600, rvu->rvu_dbg.nix,
-				    rvu, &rvu_dbg_nix_tx_stall_hwissue_fops);
-	if (!pfile)
-		goto create_failed;
+	if (is_rvu_96xx_A0(rvu)) {
+		pfile = debugfs_create_file("tx_stall_hwissue", 0600,
+					    rvu->rvu_dbg.nix, rvu,
+					    &rvu_dbg_nix_tx_stall_hwissue_fops);
+		if (!pfile)
+			goto create_failed;
+	}
 
 	pfile = debugfs_create_file("qsize", 0600, rvu->rvu_dbg.nix, rvu,
 				    &rvu_dbg_nix_qsize_fops);
-- 
2.17.1

